### PR TITLE
Added final keyword adder for parameters

### PR DESCRIPTION
**Pull request by sch-cbavel**

### CODE REVIEW

File: src/test/java/org/walkmod/findbugs/transformations/AddFinalKeywordVisitorTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+public class AddFinalKeywordVisitorTest {
+    @Test
+    public void shouldAddFinalKeywordToParameters() throws Exception {
+        final String base = "final-parameters/without-final";
+        final String input = base + "/input";
+        final String expected = base + "/expected";
+        final CompilationUnit cu = ASTManager.parse(getResourceFile(input));
+        final AddFinalKeywordVisitor visitor = new AddFinalKeywordVisitor();
+        visitor.visit(cu, null);
+        Assert.assertEquals(getResourceString(expected), cu.toString());
```
Comment (yuriytkach) [inline src/test/java/org/walkmod/findbugs/transformations/AddFinalKeywordVisitorTest.java]: The only comment - let's use assertj everywhere to be consistent :)

**Review by yuriytkach — APPROVED** [review]
Minor comment. Other than that - great!

File: src/test/java/org/walkmod/findbugs/transformations/AddFinalKeywordVisitorTest.java (hunk 1 above)
Comment (sch-cbavel) [reply src/test/java/org/walkmod/findbugs/transformations/AddFinalKeywordVisitorTest.java]: Done :) Never used assertj

**Comment by sch-cbavel** [comment]
@yuriytkach I have added a commit where I just add final keyword if the method do not modify the parameter.

Maybe we can emit a warning somehow in the cases we are modifying the parameters and advise that it is a bad practice.

**Comment by yuriytkach** [comment]
Maybe we can add a comment with wallkmod, saying like "FIXME: Do not modify parameter!" :)

**Comment by yuriytkach** [comment]
Oh, and looks like you need to run that visitor on the your class itself :stuck_out_tongue_winking_eye: (looks like you don't have final's on your params)

**Comment by sch-cbavel** [comment]
I'm not used to Java :) Most of my spare time I write Kotlin and there you cannot modify arguments :P

**Comment by yuriytkach** [comment]
Great!
Let's add a comment to the output code, so if user modifies parameter, the code looks like that:
```
class Demo {
     public static void demo(String a, int b, int c) {
         for (b = 0; b < c; b++) {
            // FIXME: Do not modify parameter!
             a += "" + b;
         }
     }
 }

```

**Comment by sch-cbavel** [comment]
@rpau Do you know how to add a comment to a statement? I'm just missing that part.

**Comment by sch-cbavel** [comment]
Also I really suspect that what I'm doing probably somebody has done it already before :)

**Comment by sch-cbavel** [comment]
@yuriytkach done! :)

**Comment by sch-cbavel** [comment]
BTW. I cannot merge, so go ahead if you find how it is right now :+1:

**Pull request merged**